transcom/mymove
# B 21437 int

## [B-21437](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1015268&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## Summary

For office users reviewing Orders, there has been a false assumption that options listed for **department indicator** will match up to our backend models for **service member affiliation**. This backlog attempts to remedy this false assumption by introducing an API enum and backend type for DepartmentIndicator. 

By doing so, it resolves the API Validation errors causing issues for orders that have department indicators of Army Corps of Engineers or Department of Secretary of Defense, as they have no corresponding backend affiliation. Additionally, it allows us to remove the coupled "Navy and Marines" as well as "Air Force and Space Force" affiliation definitions that do not have true affiliation values.

Additionally, this change does so while not changing the LOA lookup mechanics, as the affiliation/department indicator is only used to filter things for Coast Guard orders.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Log in as a Service Counselor. At the end of the testing, repeat these instructions as a TOO as well to ensure that their frontend is also setup correctly.
2. Navigate to any move in your counselor's queue.
3. View the move's orders.
4. Here's where things get tricky: In order for this page to request an LOA, there are 3 pieces of data that need to be sent off. They are:
       a. Orders Issue Date
       b. Department Indicator
       c. TAC
5. Primarily for this task, we want to verify that we do not receive an error when providing a valid orders issue date, any department indicator, and a valid TAC. Try all department indicators while looking at your network log and ensure that there are no errors for any of the options. If the response is blank, but doesn't include an error, that's okay. Before this task, submitting with Army Corps of Engineers or Department of Secretary of Defense would return an unprocessable entity error - this is what we're trying to ensure is resolved with this step.
6. Now to test and ensure that the API is still functioning properly. For this, we'll need to ensure the database has a valid TAC and LOA that we can hit. There's more reading on this on the int PRs for [B-19890](https://github.com/transcom/mymove/pull/13003) and [B-19891](https://github.com/transcom/mymove/pull/13070) , but to try to keep things simple: For a TAC to return an LOA, the correct TAC obviously needs to be used for the input. The Orders Issue Date must be within both the TAC and LOA start/end dates. The TAC must have the LOA ID in the loa_id column. They must also have identical tac_sys_id and loa_sys_id. And finally, for the LOA to be valid, it must have all of the options filled per [these changes](https://github.com/transcom/mymove/pull/13070/files#diff-d0155f263ddb2dd6f7e73a634145e822e5717565dd51f10f52ecf9845293f57b) . It was easiest for me to modify entries directly in the database for testing.
7. Once verified that LOAs are retrieved properly, as well as proper API status codes being returned for errors, we can be sure that this didn't introduce any major breaking changes while also solving the issue of the validation errors.


### Backend


## Screenshots
